Spanner/tests/System/QueryTest.php

```diff
@@ -608,4 +668,396 @@ public function arrayTypesNull()
+                'structParam' => []
+            ],
+            'types' => [
+                'structParam' => new StructType
+            ]
+        ]);
+
+        $row = $res->rows()->current();
+
+        $this->assertFalse($row[0]);
+    }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindStructNoFieldsParameter()
+    {
+        $db = self::$database;
+        $res = $db->execute('SELECT @structParam IS NULL', [
+            'parameters' => [
+                'structParam' => null
+            ],
+            'types' => [
+                'structParam' => new StructType
+            ]
+        ]);
+
+        $row = $res->rows()->current();
+
+        $this->assertTrue($row[0]);
+    }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindStructParameterNullFields()
+    {
+        $db = self::$database;
+        $res = $db->execute('SELECT @structParam.f1', [
+            'parameters' => [
+                'structParam' => [
+                    'f1' => null
+                ]
+            ],
+            'types' => [
+                'structParam' => (new StructType)
+                    ->add('f1', Database::TYPE_INT64)
+            ]
+        ]);
+
+        $row = $res->rows()->current();
+        $this->assertNull($row['f1']);
+    }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindStructParameterEqualityCheck()
+    {
+        $db = self::$database;
+        $res = $db->execute('SELECT @structParam = STRUCT<threadf INT64, userf STRING>(1, "bob")', [
+            'parameters' => [
+                'structParam' => [
+                    'threadf' => 1,
+                    'userf' => 'bob',
+                ]
+            ],
+            'types' => [
+                'structParam' => (new StructType)
+                    ->add('threadf', Database::TYPE_INT64)
+                    ->add('userf', Database::TYPE_STRING)
+            ]
+        ]);
+
+        $row = $res->rows()->current();
+    }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindStructParameterNullCheck()
+    {
+        $db = self::$database;
+        $res = $db->execute('SELECT @structParam IS NULL', [
+            'parameters' => [
+                'structParam' => [
+                    'userf' => 'bob',
+                    'threadf' => 1
+                ]
+            ],
+            'types' => [
+                'structParam' => (new StructType)
+                    ->add('userf', Database::TYPE_STRING)
+                    ->add('threadf', Database::TYPE_INT64)
+            ]
+        ]);
+
+        $row = $res->rows()->current();
+        $this->assertFalse($row[0]);
+    }
+
+    /**
+     * @group spanner-query-struct
+     */
+    public function testBindArrayOfStructsParameter()
+    {
+        $db = self::$database;
+        $res = $db->execute('SELECT a.threadid FROM UNNEST(@structParam.arraysf) a', [
```

Comment:
you dont have to nest arraysf inside structParam

Reply:
This looks like the same thing you mentioned on google-cloud-ruby (https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2067#discussion_r185628406). Is it something I need to resolve?

Reply:
No this is fine.